### ISSUE TITLE
latest version 9.1.0 gives error when fs-extra is used in main process of electron while making a build.

### ISSUE THREAD

**Issue by rishabbhattacharyya** [jprichardson/node-fs-extra#863]
- **Operating System:** Windows
- **Node.js version:** 12.18.3
- **`fs-extra` version:** 9.1.0

**Comment by RyanZim** [comment]
@rishabbhattacharyya please post the full error message, with stack trace.

**Comment by rishabbhattacharyya** [comment]
hi here is my full error message with the stack trace



PS C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app> yarn make
yarn run v1.22.4
$ electron-forge make
√ Checking your system
√ Resolving Forge Config
We need to package your application before we can make it
| Compiling Main Process CodeBrowserslist: caniuse-lite is outdated. Please run the following command: `yarn upgrade`
× Compiling Main Process Code

An unhandled error has occurred inside Forge:
Compilation errors in the main process: Hash: 37d96167171647e27c3e
Version: webpack 4.44.2
Time: 4141ms
Built at: 01/26/2021 11:07:48 PM
                                   Asset     Size  Chunks         Chunk Names
                                index.js  108 KiB       0         main
                            index.js.map  388 KiB       0  [dev]  main
native_modules/build\Release\keytar.node  617 KiB
Entrypoint main = index.js index.js.map
  [0] external "electron" 42 bytes {0} [built]
  [1] external "path" 42 bytes {0} [built]
  [5] ./node_modules/core-js/internals/fails.js 108 bytes {0} [built]
 [43] ./node_modules/core-js/modules/es.regexp.exec.js 186 bytes {0} [built]
 [53] external "child_process" 42 bytes {0} [built]
 [60] multi ./src/main.ts 28 bytes {0} [built]
 [61] ./src/main.ts 8.13 KiB {0} [built]
 [62] ./node_modules/core-js/modules/es.array.join.js 686 bytes {0} [built]
 [83] ./node_modules/core-js/modules/es.string.replace.js 4.96 KiB {0} [built]
 [87] ./node_modules/core-js/modules/es.string.split.js 5.33 KiB {0} [built]
 [91] ./node_modules/rimraf/rimraf.js 8.66 KiB {0} [built]
[101] ./node_modules/electron-squirrel-startup/index.js 1 KiB {0} [built]
[109] ./node_modules/electron-windows-badge/index.js 1.35 KiB {0} [built]
[111] ./node_modules/keytar/lib/keytar.js 1.01 KiB {0} [built]
[113] ./node_modules/fs-extra/lib/index.js 719 bytes {0} [built]
    + 125 hidden modules

ERROR in ./node_modules/fs-extra/lib/ensure/symlink-type.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (22:10)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$1.parseFunction (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19273:8)
    at Object.module.exports.pp$1.parseFunctionStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18974:15)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18826:17)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseTopLevel (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18739:23)
    at Object.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18542:15)
    at Function.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18565:35)
 @ ./node_modules/fs-extra/lib/ensure/symlink.js 14:21-46
 @ ./node_modules/fs-extra/lib/ensure/index.js
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/remove/rimraf.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (293:14)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18841:34)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseDoStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18907:20)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18819:31)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18841:34)
 @ ./node_modules/fs-extra/lib/remove/index.js 4:15-34
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/empty/index.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (33:10)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$1.parseFunction (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19273:8)
    at Object.module.exports.pp$1.parseFunctionStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18974:15)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18826:17)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseTopLevel (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18739:23)
    at Object.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18542:15)
    at Function.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18565:35)
 @ ./node_modules/fs-extra/lib/index.js 9:5-23
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/ensure/file.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (47:10)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$1.parseFunction (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19273:8)
    at Object.module.exports.pp$1.parseFunctionStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18974:15)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18826:17)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseTopLevel (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18739:23)
    at Object.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18542:15)
    at Function.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18565:35)
 @ ./node_modules/fs-extra/lib/ensure/index.js 3:13-30
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/mkdirs/make-dir.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (85:14)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19073:24)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$3.parseArrowExpression (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20638:8)
    at Object.module.exports.pp$3.parseExprAtom (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20227:21)
    at Object.parseExprAtom (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4551:117)
 @ ./node_modules/fs-extra/lib/mkdirs/index.js 3:43-64
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts
Error: Compilation errors in the main process: Hash: 37d96167171647e27c3e
Version: webpack 4.44.2
Time: 4141ms
Built at: 01/26/2021 11:07:48 PM
                                   Asset     Size  Chunks         Chunk Names
                                index.js  108 KiB       0         main
                            index.js.map  388 KiB       0  [dev]  main
native_modules/build\Release\keytar.node  617 KiB
Entrypoint main = index.js index.js.map
  [0] external "electron" 42 bytes {0} [built]
  [1] external "path" 42 bytes {0} [built]
  [5] ./node_modules/core-js/internals/fails.js 108 bytes {0} [built]
 [43] ./node_modules/core-js/modules/es.regexp.exec.js 186 bytes {0} [built]
 [53] external "child_process" 42 bytes {0} [built]
 [60] multi ./src/main.ts 28 bytes {0} [built]
 [61] ./src/main.ts 8.13 KiB {0} [built]
 [62] ./node_modules/core-js/modules/es.array.join.js 686 bytes {0} [built]
 [83] ./node_modules/core-js/modules/es.string.replace.js 4.96 KiB {0} [built]
 [87] ./node_modules/core-js/modules/es.string.split.js 5.33 KiB {0} [built]
 [91] ./node_modules/rimraf/rimraf.js 8.66 KiB {0} [built]
[101] ./node_modules/electron-squirrel-startup/index.js 1 KiB {0} [built]
[109] ./node_modules/electron-windows-badge/index.js 1.35 KiB {0} [built]
[111] ./node_modules/keytar/lib/keytar.js 1.01 KiB {0} [built]
[113] ./node_modules/fs-extra/lib/index.js 719 bytes {0} [built]
    + 125 hidden modules

ERROR in ./node_modules/fs-extra/lib/ensure/symlink-type.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (22:10)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$1.parseFunction (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19273:8)
    at Object.module.exports.pp$1.parseFunctionStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18974:15)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18826:17)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseTopLevel (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18739:23)
    at Object.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18542:15)
    at Function.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18565:35)
 @ ./node_modules/fs-extra/lib/ensure/symlink.js 14:21-46
 @ ./node_modules/fs-extra/lib/ensure/index.js
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/remove/rimraf.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (293:14)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18841:34)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseDoStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18907:20)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18819:31)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18841:34)
 @ ./node_modules/fs-extra/lib/remove/index.js 4:15-34
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/empty/index.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (33:10)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$1.parseFunction (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19273:8)
    at Object.module.exports.pp$1.parseFunctionStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18974:15)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18826:17)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseTopLevel (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18739:23)
    at Object.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18542:15)
    at Function.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18565:35)
 @ ./node_modules/fs-extra/lib/index.js 9:5-23
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/ensure/file.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (47:10)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$1.parseFunction (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19273:8)
    at Object.module.exports.pp$1.parseFunctionStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18974:15)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18826:17)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseTopLevel (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18739:23)
    at Object.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18542:15)
    at Function.parse (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18565:35)
 @ ./node_modules/fs-extra/lib/ensure/index.js 3:13-30
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts

ERROR in ./node_modules/fs-extra/lib/mkdirs/make-dir.js
Module build failed (from ./node_modules/@marshallofsound/webpack-asset-relocator-loader/dist/index.js):
SyntaxError: Unexpected token (85:14)
    at Object.module.exports.pp$4.raise (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20834:13)
    at Object.module.exports.pp.unexpected (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18680:8)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19069:49)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$1.parseTryStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19073:24)
    at Object.module.exports.pp$1.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:18834:32)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4539:118)
    at Object.parseStatement (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:42314:22)
    at Object.module.exports.pp$1.parseBlock (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:19157:23)
    at Object.module.exports.pp$3.parseFunctionBody (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20675:22)
    at Object.module.exports.pp$3.parseArrowExpression (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20638:8)
    at Object.module.exports.pp$3.parseExprAtom (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:20227:21)
    at Object.parseExprAtom (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@marshallofsound\webpack-asset-relocator-loader\dist\index.js:4551:117)
 @ ./node_modules/fs-extra/lib/mkdirs/index.js 3:43-64
 @ ./node_modules/fs-extra/lib/index.js
 @ ./src/main.ts
 @ multi ./src/main.ts
    at cb (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\@electron-forge\plugin-webpack\src\WebpackPlugin.ts:250:31)
    at finalCallback (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\webpack\lib\Compiler.js:257:39)
    at C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\webpack\lib\Compiler.js:273:13
    at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:4:1)
    at AsyncSeriesHook.lazyCompileHook (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\Hook.js:154:20)
    at onCompiled (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\webpack\lib\Compiler.js:271:21)
    at C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\webpack\lib\Compiler.js:681:15
    at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:4:1)
    at AsyncSeriesHook.lazyCompileHook (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\Hook.js:154:20)
    at C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\webpack\lib\Compiler.js:678:31
    at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:4:1)
    at AsyncSeriesHook.lazyCompileHook (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\Hook.js:154:20)
    at C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\webpack\lib\Compilation.js:1423:35
    at AsyncSeriesHook.eval [as callAsync] (eval at create (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:4:1)
    at AsyncSeriesHook.lazyCompileHook (C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\tapable\lib\Hook.js:154:20)
    at C:\Users\liris\Desktop\project\TypeScript_Version\vzm-desktop-app\node_modules\webpack\lib\Compilation.js:1414:32
error Command failed with exit code 1.

**Comment by RyanZim** [comment]
Looks like there's something in your toolchain that's outdated enough not to support optional catch bindings, which fs-extra uses; hence the error. Look into upgrading webpack/electron-forge/etc.

**Comment by mcunningham** [comment]
@rishabbhattacharyya did you find a solution to this? I have the same issue

**Comment by rishabbhattacharyya** [comment]
> @rishabbhattacharyya did you find a solution to this? I have the same issue

I changed the version of fs-extra in package.json from 9.1.0 to 8.0.0, which removed the error. As I was using some outdated libraries which fs-extra 9.1.0 would support.

**Comment by ivancuric** [comment]
Same issue here. Using `fs-extra` in the render process directly, or via preload doesn't work. Latest working version was 8.0.0.

**Comment by RyanZim** [comment]
@ivancuric Please open a new issue with the full error, electron version, etc.
